naga/src/front/wgsl/error.rs

```diff
@@ -820,6 +822,31 @@ impl<'a> Error<'a> {
                     "perhaps you meant to call the function in a separate statement?".into(),
                 ],
             },
+            Error::FunctionMustUseUnused(call) => ParseError {
+                message: "must use result".into(),
+                labels: vec![(call, "result unused".into())],
+                notes: vec![
+                    format!(
+                        "function '{}' is declared with `@must_use` attribute",
+                        &source[call].split_once('(').unwrap().0
+                    ),
+                    "call the function as part of an expression".into(),
+                ],
+            },
+            Error::FunctionMustUseNothing(attr, signature) => ParseError {
+                message: "must use nothing".into(),
+                labels: vec![
+                    (attr, "requires result".into()),
+                    (signature, "no result".into()),
+                ],
+                notes: vec![
+                    format!(
+                        "function '{}' is declared with `@must_use` attribute",
+                        &source[signature].split_once('(').unwrap().0
+                    ),
+                    "declare a return type or remove the attribute".into(),
```

Comment:
And for this one:
- message could be eg: "function annotated with @must_use but does not return any value"
- again I don't think the text labels actually add anything, just highlighting the spans is good enough, so would use empty strings for the labels
- I don't think the first note adds anything as the message and label already draw attention to that. the second note is great